src/fs/hardlinkDir.ts

```diff
@@ -9,8 +9,15 @@ export default async function hardlinkDir(existingDir: string, newDir: string) {
     dirs
       .map(async (relativePath: string) => {
         const existingPath = path.join(existingDir, relativePath)
-        const newPath = path.join(newDir, relativePath);
-        const stat = await fs.stat(existingPath)
+        const newPath = path.join(newDir, relativePath)
+        const stat = await fs.lstat(existingPath)
+        if (stat.isSymbolicLink()) {
+          // filter out broken symlinks
+          if (!await fs.exists(existingPath)) {
```

Comment:
if this is an error, maybe should log a warning? You can `import logger from 'pnpm-logger'` and use the `logger.warn` function